src/main/java/com/riwi/localstorage/riwi_local_storage/infrastructure/services/DiscountService.java

```diff
@@ -31,8 +31,9 @@ public class DiscountService implements IDiscountService {
 
   @Override
   public DiscountResponse create(DiscountRequest request) {
```

Comment:
create method is missing the code validation, discount entity should not have repeated code. 
Use validateDiscountRequest method below.